### ISSUE TITLE
Some problems

### ISSUE THREAD

**Issue by drogen120** [LiamLYJ/scene_seg#1]
1. some kernel size is different from the paper.
In "4.1 Encoding network" of the paper, they said "we use layer conv5_1 as an input to a residual block, consisting of three
convolutional layers with 3 × 3 kernels (the first two of which are followed by ReLU non-linearities,
while the third one is linear),"
So the network should be:

> 

    def __init__(self, inplanes, planes, stride=1, downsample=None):
        super(Bottleneck, self).__init__()
        self.conv1 = conv1x1(inplanes, planes) #should be 3x3
        self.bn1 = nn.BatchNorm2d(planes)
        self.conv2 = conv3x3(planes, planes, stride)
        self.bn2 = nn.BatchNorm2d(planes)
        self.conv3 = conv1x1(planes, planes * self.expansion) #should be 3x3
        self.bn3 = nn.BatchNorm2d(planes * self.expansion)
        self.relu = nn.ReLU(inplace=True)
        self.downsample = downsample
        self.stride = stride

2. You used AdaptiveAvgPool2d to generate the texture features. It is good for some big textures, but I am afraid it difficult to get good results for some fine textures.

> 

    filt_adp = nn.AdaptiveAvgPool2d((5,5))
    filt = filt_adp(output_encoder_patch)
    correlations = nn_F.conv2d(output_encoder_img, filt, stride = 1, padding = 2)

**Comment by LiamLYJ** [comment]
thanks a lot 
I changed the code of 1x1 to 3x3 in res_block 
by the way, 64x64 filter size may convolve a lot of computational cost
